src/front/glsl/variables.rs

```diff
@@ -21,19 +21,12 @@ impl Program {
         match name {
             "gl_Position" => {
                 #[cfg(feature = "glsl-validate")]
-                match self.shader_stage {
-                    ShaderStage::Vertex | ShaderStage::Fragment { .. } => {}
-                    _ => {
-                        return Err(ErrorKind::VariableNotAvailable(name.into()));
-                    }
+                if !self.has_stage(ShaderStage::Vertex) {
```

Comment:
I don't think we ever need to call `has_stage`. Why are we checking it? If no entry points use this global variable, it's totally fine for it to stay in the IR.

Reply:
I think this was done in order to be able to give errors, if e.g. a fragment shader references a builtin that is only available in vertex stage. But when supporting multiple stages this gets less clear.

Reply:
This is caught by IR validation already, you don't need to to worry about this in the frontend.